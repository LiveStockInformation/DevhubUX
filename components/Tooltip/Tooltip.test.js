import * as React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Tooltip } from '.'

afterEach(cleanup)

describe('Tooltip component', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Tooltip tip='Test tooltip' />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('shows a tooltip on mouse hover', () => {
    const { container } = render(<Tooltip tip='Test tooltip' />)
    const tooltipButton = container.querySelector('.tooltip__btn')
    const tooltipPopper = container.querySelector('.tooltip__popper')
    fireEvent.mouseOver(tooltipButton)
    expect(tooltipPopper).toHaveClass('tooltip__popper--active')
  })
  it('hides tooltip on mouse leave', () => {
    const { container } = render(<Tooltip tip='Test tooltip' />)
    const tooltipButton = container.querySelector('#tooltip')
    const tooltipPopper = container.querySelector('.tooltip__popper')
    fireEvent.mouseOver(tooltipButton)
    fireEvent.mouseOut(tooltipButton)
    expect(tooltipPopper).toHaveClass('tooltip__popper')
  })
  it('shows a tooltip on focus hover', () => {
    const { container } = render(<Tooltip tip='Test tooltip' />)
    const tooltipButton = container.querySelector('.tooltip__btn')
    const tooltipPopper = container.querySelector('.tooltip__popper')
    fireEvent.focus(tooltipButton)
    expect(tooltipPopper).toHaveClass('tooltip__popper--active')
  })
  it('hides tooltip on blur', () => {
    const { container } = render(<Tooltip tip='Test tooltip' />)
    const tooltipButton = container.querySelector('.tooltip__btn')
    const tooltipPopper = container.querySelector('.tooltip__popper')
    fireEvent.focus(tooltipButton)
    fireEvent.blur(tooltipButton)
    expect(tooltipPopper).toHaveClass('tooltip__popper')
  })
  it('has absolute positioning style, with absolute prop', () => {
    const { container } = render(
      <Tooltip tip='Test tooltip' positionAbsoluteLeft />
    )
    const tooltip = container.querySelector('.tooltip')
    expect(tooltip).toHaveClass('tooltip--absolute')
  })
})
