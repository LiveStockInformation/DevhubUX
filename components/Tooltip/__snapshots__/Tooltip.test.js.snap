// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="tooltip  "
  >
    <button
      aria-label="Tooltip"
      aria-labelledby="tooltip"
      class="tooltip__btn"
    >
      <span
        class="visually-hidden"
      >
        Helpful Tooltip
      </span>
    </button>
    <div
      class="tooltip__popper "
      id="tooltip"
      role="tooltip"
    >
      <div
        class="tooltip__popper-content"
      >
        Test tooltip
      </div>
    </div>
  </div>
</DocumentFragment>
`;
